### DISCUSSION TITLE
Deployment - Build classes from class definitions

### DISCUSSION THREAD

**Discussion by Anaxarchos** [pimcore/pimcore#13191] [Q&A / Support]
Hi,
I just pulled some class definitions into var/classes/ and then executed 
`bin/console pimcore:build:classes`

After that I have the desired class in var/classes/DataObject and var/classes/DataObject/MyClass/Listing.php

But the DataObject does not show up in the backend.
The cache was cleared with 
`bin/console cache:clear` 
and I cleared the browser cache as well.

Thank you for your help.

**Comment by Anaxarchos** [comment]
A short look in the DB showed me, that the DataObjects were not added to the classes column.

**Reply by jdreesen** [reply]
`pimcore:build:classes` only generates the PHP code. You have to run `pimcore:deploy:classes-rebuild` for the DB changes.

**Reply by Anaxarchos** [reply]
Is it
`pimcore:deploy:classes-rebuild`
or 
`pimcore:deployment:classes-rebuild`

**Reply by jdreesen** [reply]
I think it's the latter, but it should work both, as Symfony allows to abbreviate commands.

**Reply by Anaxarchos** [reply]
Sorry, 
I forgot to mention, that I did this already once, as suggested in this post: https://github.com/pimcore/pimcore/discussions/13071#discussioncomment-3603868
And back then it worked fine. Now, after a new pull request, and after you reminded me of using that command, I get a 504 gateway timeout error message and no hint in the dev.log file. 
On my local machine the pimcore app is running without any errors. 
And I ran a `composer update` before running `pimcore:deployment:classes-rebuild`

**Reply by Anaxarchos** [reply]
Now, after restarting the server and installing a new version of pimcore,
I did a pull request, then I did a `composer update`, then executed the `bin/console pimcore:deployment:classes-rebuild` and though I find all the definitions in var/classes and all the DataObjects and fieldcollections where they should be: nothing happens. 
The command just runs for a second without throwing an error and without any result.
Thank you in advance.

**Reply by jdreesen** [reply]
So the classes are generated in `var/classes/DataObject` but not in the database? Did you run it with `--create-classes`? That's necessary for them to be created in the database, too.

**Reply by Anaxarchos** [reply]
Thanx, this works fine.
